Assets/Scripts/Edges/LineFactory.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System.Collections.Generic;
+using Assets.Scripts.Common.Utils;
+using UnityEngine;
+
+namespace Assets.Scripts.Edges
+{
+  public class LineFactory : MonoBehaviour
+  {
+    public BezierCurveFactory bezierCurveFactory;
+    public bool debug;
+    public DebugSphereFactory debugSphereFactory;
+
+    /// <summary>
+    ///   Returns a list of points that make up a curved, rotated edge when <paramref name="numberOfLines" /> > 1
+    ///   and a straight, two points line when it is == 1
+    /// </summary>
+    public IList<IList<Vector3>> GetLinePositionsFor(Vector3 startingPoint, Vector3 endingPoint, int numberOfLines)
+    {
+      switch (numberOfLines)
+      {
+        case 0:
+          return new List<IList<Vector3>>();
+        case 1:
+          return GetEdgePointsForSingleEdge(startingPoint, endingPoint);
+        default:
+          return GetEdgePointsForMultipleEdges(startingPoint, endingPoint, numberOfLines);
+      }
+    }
+
+    IList<IList<Vector3>> GetEdgePointsForSingleEdge(Vector3 startingPoint, Vector3 endingPoint) =>
+      new List<IList<Vector3>> {new List<Vector3> {startingPoint, endingPoint}};
+
+    IList<IList<Vector3>> GetEdgePointsForMultipleEdges(Vector3 startingPoint, Vector3 endingPoint, int numberOfLines)
+    {
+      var rotation = 360f / numberOfLines;
+      var currentRotation = 0f;
+      var middlePoint = (endingPoint + startingPoint) / 2;
+      var rotationAxis = (endingPoint - startingPoint).normalized;
+      var normalVector = Vector3.Cross(Vector3.up, rotationAxis).normalized;
+
+      if (debug)
+      {
+        debugSphereFactory.ClearDebugSpheres();
+        debugSphereFactory.AddDebugSphere(middlePoint);
+      }
+
+      var curvePointsForEdges = new List<IList<Vector3>>();
+
+      for (var _ = 0; _ < numberOfLines; _++)
+      {
+        curvePointsForEdges.Add(
+          CreateCurvedEdgeGameObject()
+        );
+        currentRotation += rotation;
+      }
+
+      return curvePointsForEdges;
+
+      IList<Vector3> CreateCurvedEdgeGameObject()
```

Comment:
Nested methods are a bit discouraged, instead pure methods should be used. I can't easily read this code and know what's happening as the nested method uses variables from the outer one and it's scrolling back and forth.

It's alright for now, but consider dropping those in the future.